src/Controller/userController.js

```diff
@@ -0,0 +1,145 @@
+const axios = require("axios");
+const User = require("../Models/User");
+const env = require("../configs/dev");
+
+const login = async () => {
+  // res.json(req.user);
+};
+
+const loginCallback = async (req, res) => {
+  res.json({ user: req.user });
+};
+const registerCallback = async (req, res) => {
+  res.json({ user: req.user });
+};
+const getAllUsers = async (req, res, next) => {
+  try {
+    User.find()
+      .populate("role")
+      .populate({
+        path: "role",
+        populate: {
+          path: "permissions",
+          model: "Permissions",
+        },
+      })
+      .then(
+        (users) => {
+          res
+            .status(200)
+            .json({ success: true, message: "Users found", data: users });
+        },
+        (err) => next(err)
+      );
+  } catch (err) {
+    next(err);
+  }
+};
+const getUserById = async (req, res, next) => {
+  try {
+    User.findById(req.params.id)
+      .populate("role")
+      .populate({
+        path: "role",
+        populate: {
+          path: "permissions",
+          model: "Permissions",
+        },
+      })
+      .then(
+        (user) => {
+          res
+            .status(200)
+            .json({ success: true, message: "User found", data: user });
+        },
+        (err) => next(err)
+      );
+  } catch (err) {
+    next(err);
+  }
+};
+
+const createUser = async (req, res, next) => {
+  const user = new User(req.body);
+
+  try {
+    User.create(user)
+      .then(
+        (newUser) => {
+          res
```

Comment:
I've noticed the inconsistency in your response objects e.g.

```js
// create role res
res.status(200).json(role);

// create user res
res.status(201).json({ success: true, message: "User created", data: newUser });


One has a success property the other doesn't. Kindly have consistent responses across the entire codebase. Also add error responses with the appropriate error http status codes.

Reply:
I have been using correct error codes, but sometimes I just failed to notice thanks for correcting me out.